### PR TITLE
Eliminate exception-message substring checks for unknown entity types

### PR DESCRIPTION
**Pull request by Pr0methean**
Instead, use a new exception class for that case. Also i18nize the relevant strings.

### CODE REVIEW

File: src/main/java/net/glowstone/io/entity/UnknownEntityTypeException.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+     * Returns the ID subtag for this entity, if it has one that's a string; otherwise, returns the
+     * entire NBT tag as a string.
+     *
+     * @return the ID, or the entire NBT tag as a string if no ID is defined
+     */
+    public String getIdOrTag() {
+        return nbt.isString("id") ? nbt.getString("id") : nbt.toString();
+    }
+
+    public UnknownEntityTypeException(CompoundTag nbt) {
```
Comment (aramperes) [inline src/main/java/net/glowstone/io/entity/UnknownEntityTypeException.java]: The constructor should be at the top of the class

File: src/main/java/net/glowstone/io/entity/UnknownEntityTypeException.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
+package net.glowstone.io.entity;
+
+import lombok.Getter;
+import net.glowstone.i18n.LocalizedStrings;
+import net.glowstone.util.nbt.CompoundTag;
+
+public class UnknownEntityTypeException extends IllegalArgumentException {
```
Comment (aramperes) [inline src/main/java/net/glowstone/io/entity/UnknownEntityTypeException.java]: Could you add documentation for this exception type?

**Review by aramperes — DISMISSED** [review]

**Comment by aramperes** [comment]
Could you fix the merge conflicts?

**Comment by Pr0methean** [comment]
Done.

**Pull request merged**